src/components/Todo/Todo.tsx

```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDelete: (id: number) => Promise<void>;
+  isAdding?: boolean;
+  clearCompleted: boolean;
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  onDelete,
+  isAdding,
+  clearCompleted,
+}) => {
+  const { completed, title, id } = todo;
+  const [isLoading, setIsLoading] = useState(false);
+  const remove = () => {
+    setIsLoading(true);
+    onDelete(id)
+      .finally(() => setIsLoading(false));
+  };
+
+  useEffect(() => {
+    if (clearCompleted && completed) {
+      remove();
+    }
+  }, [clearCompleted]);
```

Comment:
Provide all dependencies
```suggestion
  useEffect(() => {
    if (clearCompleted && completed) {
      remove();
    }
  }, [clearCompleted, completed]);
```